tests/multiple_listening_addresses.rs

```diff
@@ -30,8 +30,22 @@ fn multiple_concurrent_ephemeral_listening_addresses_on_same_ip() {
 
         let (first, second) = futures::future::join(first, second).await;
 
-        // one should succeed and other should fail
-        assert_eq!(first.is_ok(), second.is_err());
+        // before an Swarm alike api on the background task to make sure we attempt to modify the
```

Comment:
nit(grammar): "...a Swarm-like api in the background..."; also wording - the message got mangled here

Reply:
Trying to rework this with more context.